sale_purchase_history/views/product_sale_history_views.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<odoo>
+    <record id='product_sale_history_action' model="ir.actions.act_window">
+        <field name="name">Product Sale/Purchase History</field>
+        <field name="res_model">product.sale.history</field>
+        <field name="view_mode">form</field>
+    </record>
+
+    <record id='product_sale_history_form' model='ir.ui.view'>
+        <field name='name'>product.sale.history.form</field>
+        <field name='model'>product.sale.history</field>
+        <field name='arch' type='xml'>
+            <form>
+                <field name="is_purchase" invisible="True" />
+                <field name="is_sale" invisible="True" />
+                <group>
+                    <field name="product_id" />
+                </group>
+                <field name="sale_order_lines" invisible="is_purchase" />
+                <field name="purchase_order_lines" invisible="is_sale" />
```

Comment:
you could use invisible as such to reduce complexity and eliminate the fields  if the goal of the fields was only visibility
<field name="sale_order_lines" invisible="not sale_order_lines" />

Reply:
I think this is better imho for readbility